src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1104,7 +1104,7 @@ string PrintProcesses()
                     builder.AppendLine();
                 }
                 
-                builder.AppendFormat("Current process id: {0}", Process.GetCurrentProcess().Id);
+                builder.AppendFormat("Current process id: {0} Process name: '{1}' Id: {2}", Process.GetCurrentProcess().Id, currentProcess.ProcessName, currentProcess.Id);
```

Comment:
It's really weird to pass in currentProcess and then use that for some things and Process.GetCurrentProcess() for others.  Just use currentProcess.Id here rather than using Process.GetCurrentProcess().Id.

Reply:
Yes, I know, bug looks weird to me also, so I added more logging as possible, here I would be sure that there are no issues with name and id of current read process used in process lookup.
Test call `Process currentProcess = Process.GetCurrentProcess();` and after assertion calls `ProcessName` that is loaded throught `GetProcessInfo` using PID

https://github.com/dotnet/corefx/blob/689e58ec123d8c730d48d1c35c8f7ecbba6c22cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs#L80

It's a double check.
Looking at code I found strange comment like 

https://github.com/dotnet/corefx/blob/689e58ec123d8c730d48d1c35c8f7ecbba6c22cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L350

I don't know is it's possible, but in case of duplicate pid we could pickup wrong name due to override
https://github.com/dotnet/corefx/blob/689e58ec123d8c730d48d1c35c8f7ecbba6c22cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L403

Reply:
@danmosemsft @stephentoub another source of issue could be a sync issue on `static buffer` used for `Interop.NtDll.NtQuerySystemInformation` I don't see any sync around and in case of concurrent call same buffer is passed(possible in tests process) and I don't think that kernel sync in some way https://docs.microsoft.com/en-us/windows/desktop/api/winternl/nf-winternl-ntquerysysteminformation

https://github.com/dotnet/corefx/blob/689e58ec123d8c730d48d1c35c8f7ecbba6c22cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L241
https://github.com/dotnet/corefx/blob/689e58ec123d8c730d48d1c35c8f7ecbba6c22cf/src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L312

Reply:
> It's a double check.

You're essentially validating in a roundabout way that Process.GetCurrentProcess().Id == Process.GetCurrentProcess().Id.  We do not need to validate that in this test.

> but in case of duplicate pid we could pickup wrong name due to override

That is not going to affect the current process.

> I don't see any sync around and in case of concurrent call same buffer is passed

See the Interlocked.Exchange at the beginning and end.  The method uses an atomic operation to rent the buffer, nulling out the shared one at the same time, and then swaps it back in at the end.  Only one thread will be able to use this buffer at a time; everyone else will be forced to allocate one a new. (Separately we may want to look at switching to using ArrayPool for this.)

Reply:
> See the Interlocked.Exchange at the beginning and end. T...

Ah ok, so in case of concurrency we allocate more buffer and reuse only "one", thank's, messed up my mind between exchange and comparewithnullandreturnval, "my personal api" I wanted a bug here 😅

> You're essentially validating in a roundabout way that Process.GetCurrentProcess().Id == Process.GetCurrentProcess().Id. We do not need to validate that in this test.

Ok revert

> That is not going to affect the current process.

Yes unrelated to current process...I wonder if it's possible.

Reply:
done